### PR TITLE
builder feature: union

### PR DESCRIPTION
**Pull request by Outman9946**
UNION feature & tests

### CODE REVIEW

File: builder_union.go [hunk 1]
```diff
@@ -0,0 +1,29 @@
+package builder
```
Comment (lunny) [inline builder_union.go]: file head comment

File: builder_union_test.go [hunk 2]
```diff
@@ -0,0 +1,47 @@
+package builder
```
Comment (lunny) [inline builder_union_test.go]: As above

File: builder_union.go (hunk 1 above)
Comment (Outman9946) [reply builder_union.go]: added

File: builder_union_test.go (hunk 2 above)
Comment (Outman9946) [reply builder_union_test.go]: added

File: builder_union.go (hunk 1 above)
Comment (lunny) [reply builder_union.go]: 2018？

File: builder_union.go [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+		if v.builder.optype != selectType {
+			return errors.New("UNION is only allowed among SELECT operations")
+		}
+		if len(b.unions) == 1 {
+			if err := v.builder.selectWriteTo(w); err != nil {
+				return err
+			}
+		} else {
+			if idx != 0 {
+				fmt.Fprint(w, fmt.Sprintf(" union %v ", v.unionType))
```
Comment (lunny) [inline builder_union.go]: Uppercase is better. union -> UNION

**Comment by Outman9946** [comment]
I find this PR is not been merged yet. Is there anything I need to do for this PR?

**Comment by lunny** [comment]
I'm strange why the CI is always waiting. I will run test locally.

**Comment by Outman9946** [comment]
BTW, do you need UNION feature in xorm?

**Comment by lunny** [comment]
Maybe someone need but PRs are always welcome! :)

**Pull request merged**